Add telemetry support and example

alexismanin opened the pull request:
1. Activate telemetry using OpenTelemetry instrumentation
2. Add a compose configuration to test telemetry support

alexismanin committed:
fix(Docker): fix and document dev docker build

alexismanin committed:
feat(Compose): add a minimal compose configuration

alexismanin committed:
feat(Compose): Add observability stack

Add an observability configuration using Open Telemetry + Grafana to monitor application traces and logs

alexismanin committed:
feat(Python): activate uvicorn logging export using open-telemetry

alexismanin committed:
feat(Compose): Add an example dashboard for HTTP access statistics

alexismanin committed:
doc(Compose): Add documentation to help using and understanding compose configuration

Emile-Muller commented on compose/README.rst:
```diff
@@ -0,0 +1,84 @@
+
+
+.. raw:: html
+  :file: ./telemetry.drawio.html
```
Drawio file is not showing up the GitHub documentation viewer.

Emile-Muller commented on compose/README.rst:
```diff
@@ -0,0 +1,84 @@
+    #. Unzip the download file to get the *.mmdb* file directly on your system.
+#. Activate GeoIP processing
+    #. Move/copy the GeoLite2 database to `./confs/grafana/alloy/` directory, and name it `geolite2-city.mmdb`.
+    #. Uncomment the `stage.geoip` block in `Alloy configuration file <file:./confs/grafana/alloy/config.alloy>`_
```
Link to file (`config.alloy`) doesn't work.

Emile-Muller committed:
doc(compose): Changed code samples to use two backquotes instead of one (RST format).

alexismanin committed:
doc(Compose): fix image and link

alexismanin committed:
refactor(Compose): change api configuration binding

Use bind mount instead of config block, to enhance retro-compatibility with older versions of docker-compose

alexismanin committed:
fix(Python): refactor otel/uvicorn binding to use "on_event('startup')" feature instead of lifespan.

Use of lifespan broke tortoise shutdown (which use on_event("shutdown") in ws_router file.
